src/s2/util/gtl/compact_array.h

```diff
@@ -413,7 +413,9 @@ class compact_array_base {
     value_allocator_type allocator;
 
     T* new_ptr = allocator.allocate(capacity());
-    memcpy(new_ptr, Array(), old_capacity * sizeof(T));
+    if (old_capacity != 0) {
+      memcpy(new_ptr, Array(), old_capacity * sizeof(T));
+    }
     allocator.deallocate(Array(), old_capacity);
```

Comment:
Maybe better? I don't read class code

```suggestion
    if (old_capacity != 0) {
      memcpy(new_ptr, Array(), old_capacity * sizeof(T));
      allocator.deallocate(Array(), old_capacity);
    }
```